sktime/forecasting/hf_momentfm_forecaster.py

```diff
@@ -0,0 +1,532 @@
+
+    freeze_encoder : bool
+        Selection of whether or not to freeze the weights of the encoder
+        Default = True
+
+    freeze_embedder : bool
+        Selection whether or not to freeze the patch embedding layer
+        Default = True
+
+    freeze_head : bool
+        Selection whether or not to freeze the forecasting head.
+        Recommendation is that the linear forecasting head must be trained
+        Default = False
+
+    dropout : float
+        Dropout value of the model. Values range between [0.0, 1.0]
+        Default = 0.1
+
+    head_dropout : float
+        Dropout value of the forecasting head. Values range between [0.0, 1.0]
+        Default = 0.1
+
+    seq_len : int
+        length of sequences or length of historical values that are passed
+        to the model for training at each time point. the momentfm model requires
+        sequence lengths to be 512 exactly, so if less, padding will be used.
+        If the sequence length is > 512, it will be reduced to 512.
+        default = 512
+
+    forecasting_horizon : int
+        Number of time steps to forecast ahead, leave this as None if user
+        wishes to pass in a fh object instead inside the fit function
+        default = None
+
+    batch_size : int
+        size of batches to train the model on
+        default = 8
+
+    epochs : int
+        Number of epochs to fit tune the model on
+        default = 1
+
+    max_lr : float
+        Maximum learning rate that the learning rate scheduler will use
+
+    device : str
+        torch device to use gpu else cpu
+
+    train_val_split : float
+        float value between 0 and 1 to determine portions of training
+        and validation splits
+
+    transformer_backbone : str
+        d_model of a pre-trained transformer model to use. See
+        SUPPORTED_HUGGINGFACE_MODELS to specify valid models to use.
+        Default is 'google/flan-t5-large'.
+
+    config : dict, default = {}
+        If desired, user can pass in a config detailing all momentfm parameters
+        that they wish to set in dictionary form, so that parameters do not need
+        to be individually set. If a parameter inside a config is a
+        duplicate of one already passed in individually, it will be overwritten.
+    """
+
+    _tags = {
+        "scitype:y": "both",
+        "authors": ["julian-fong"],
+        "maintainers": ["julian-fong"],
+        "y_inner_mtype": "pd.DataFrame",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": True,
+        "python_dependencies": ["momentfm", "torch"],
+        "capability:global_forecasting": False,
+    }
+
+    def __init__(
+        self,
+        pretrained_model_name_or_path="AutonLab/MOMENT-1-large",
+        freeze_encoder=True,
+        freeze_embedder=True,
+        freeze_head=False,
+        dropout=0.1,
+        head_dropout=0.1,
+        seq_len=512,
+        forecasting_horizon=None,
```

Comment:
Not sure, if we should call it forecasting horizon here, since this is normally passed via fit or predict.

Reply:
So, I would propose to remove it from init and require it that it is passed to fit using the tag system.

Reply:
I've set the tags `"requires-fh-in-fit": True,`, hopefully that is enough?

Reply:
Yes, but the forecasting horizon should never be part of the init method.